peekingduck/pipeline/nodes/model/yolact_edgev1/yolact_edge_files/utils.py

```diff
@@ -0,0 +1,301 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# MIT License
+
+# Copyright (c) 2020 Haotian Liu and Rafael A. Rivera-Soto
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""A collection of utility functions used by YolactEdge
+Modifications include:
+- Removed unused python scripts in the original utils folder
+- Removed unused utility functions from the original repository
+- Merged utility functions from the layers folder
+- Refactored config file parsing
+- Modified docstrings
+"""
+
+from typing import Any, Callable, Tuple, List
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+class FastBaseTransform(torch.nn.Module):
+    """
+    Transform that does all operations on the GPU for improved speed.
+    This doesn't suppport a lot of configs and should only be used for production.
+    Maintain this as necessary.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        if torch.cuda.is_available():
+            self.mean = (
+                torch.Tensor((103.94, 116.78, 123.68))
+                .float()
+                .cuda()[None, :, None, None]
+            )
+            self.std = (
+                torch.Tensor((57.38, 57.12, 58.40)).float().cuda()[None, :, None, None]
+            )
+        else:
+            self.mean = torch.Tensor((103.94, 116.78, 123.68)).float()[
+                None, :, None, None
+            ]
+            self.std = torch.Tensor((57.38, 57.12, 58.40)).float()[None, :, None, None]
```

Comment:
Many magic numbers in above code: any idea where/why they are chosen as such?
If yes, then should document down for reference to help future devs.